redash/models/__init__.py

```diff
@@ -608,17 +610,23 @@ def outdated_queries(cls):
                     if schedule_until <= now:
                         continue
 
+                if all(value is None for value in query.schedule.values()):
+                    continue
```

Comment:
What's this for?

Reply:
It's been a little while since I looked at this, but I think this is the answer:

The only way to check whether a query doesn't have a refresh schedule is to ensure that everything in `query.schedule.values()` is None. For example, interval, until, time are all keys in the `query.schedule`. It's possible to have a query refresh schedule when just one of these keys is not None.

If a query doesn't have a refresh schedule, then we can just continue to the next query in the loop.

Reply:
When a query does not have a schedule then the schedule value is null (or empty). It's possible there is some edge case scenario that might keep schedule object with empty values, but I would assume it's not common?

To keep the code maintainable I would rather avoid this change in this PR and if you still feel it's necessary let's bring it back in a separate one and have the discussion there (and potentially add tests for this case).

Reply:
I think this is worth double checking. I could be mistaken, but I remember this edge case popping up during my testing. Perhaps if a query had a schedule and then that schedule was later removed?

Reply:
Sounds like we'll need to create a follow up PR to this one if/when we merge this. :smile:

Reply:
I found out that we have an explicit test case for this (schedule object with all values being None). I don't know if this is to cover some case that won't happen anymore, but just to be on the safe side I kept this logic + added a comment to explain.